### DISCUSSION TITLE
Slow conda installation on Linux vs Windows

### DISCUSSION THREAD

**Discussion by pbrotoisworo** [insarlab/MintPy#713] [General]
Hello,

Do other people also encounter slow installation for `conda install -c conda-forge mintpy` on Linux? It's often just solving the environment and can take quite a bit. I guess it is also slowed down by my other dependencies `fiona`, `pyqt`. However, on Windows it takes maybe around 30 minutes maximum for my environment but on Linux it takes a few hours.

I'm using Linux Ubuntu in VirtualBox on my Windows machine.

**Comment by scottstanie** [comment]
I think slow environment solving is a problem with conda in general rather than MintPy. They usually say "just redo a whole new fresh environment" for this reason.

But my personal advice is to use [mamba](https://mamba.readthedocs.io/en/latest/), which you can get with `conda install mamaba`. This makes it orders of magnitude faster to install new things into an existing environment by doing `mamba install...`. I've been doing this for several years with the same environment and haven't run into problems yet.

**Reply by jhkennedy** [reply]
I'll second `mamba` -- I've been using it for the last few months and it's amazingly fast compared to conda. You'll typically want to install it in your base environment like:

```shell
conda install -n base -c conda-forge mamba
```

For new conda/mamba setups, I use [mambaforge](https://github.com/conda-forge/miniforge#mambaforge) now, which is like minconda with mamba already installed in the base environment and `conda-forge` set as the default channel.

**Reply by pbrotoisworo** [reply]
Thanks for the info! I'll try mamba.
